query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -286,54 +297,30 @@ private String convertExpression2String(
             executionContext,
             expression.getFunction(),
             argExpression ->
-                convertExpression2String(argExpression, paramsBuilder, executionContext));
+                convertExpressionToString(argExpression, paramsBuilder, executionContext));
       case ORDERBY:
         OrderByExpression orderBy = expression.getOrderBy();
-        return convertExpression2String(orderBy.getExpression(), paramsBuilder, executionContext);
+        return convertExpressionToString(orderBy.getExpression(), paramsBuilder, executionContext);
       case VALUE_NOT_SET:
         break;
     }
     return "";
   }
 
   private String convertExpressionToMapKeyColumn(Expression expression) {
-    if ((expression.getValueCase() == COLUMNIDENTIFIER)
-        || (expression.getValueCase() == ATTRIBUTE_EXPRESSION && !isComplexAttribute(expression))) {
-      String col = viewDefinition.getKeyColumnNameForMap(getLogicalColumnName(expression));
-      if (col != null && col.length() > 0) {
-        return col;
-      }
+    String col = viewDefinition.getKeyColumnNameForMap(getLogicalColumnName(expression));
+    if (col != null && col.length() > 0) {
+      return col;
     }
-    throw new IllegalArgumentException(
-        "operator CONTAINS_KEY/KEYVALUE supports multi value column only");
+    throw new IllegalArgumentException("operator supports multi value column only");
```

Comment:
Follow up is fine but we should fixup this error message since the type check is gone. Now, this error is reached if no key column exists for the provided attribute - that could be an attribute that's mistyped, but it also may just not be a column in that view.

Reply:
So what should be the error here now ?

Reply:
`Unable to lookup key column for attribute: {}`, or something like that